src/Migrations.ts

```diff
@@ -0,0 +1,207 @@
+import * as path from 'node:path'
+import * as fsp from 'node:fs/promises'
+
+import { Database } from './Database'
+import { IMigrate } from './interfaces'
+
+import MigrationFile = IMigrate.MigrationFile
+import MigrationData = IMigrate.MigrationData
+
+const DEFAULT_MIGRATIONS_DIR_NAME = 'migrations'
+const DEFAULT_MIGRATIONS_DIR_PATH = path.join(
+  process.cwd(),
+  DEFAULT_MIGRATIONS_DIR_NAME
+)
+
+const MIGRATION_FILE_REGEXP = /^(\d+).(.*?)\.sql$/
+const MIGRATION_FILE_UP_REGEXP = /^-- .*$/gm
+const MIGRATION_FILE_DOWN_REGEXP = /^--\s+?down\b/im
+const MIGRATION_FILE_REPLACE_COMMAND_REGEXP = /^-{3,}$/gim
+
+export class Migrations<
+  Driver extends Database,
+  Config extends IMigrate.MigrationParams = IMigrate.MigrationParams
+> {
+  db: Driver
+  config: Config
+
+  constructor (db: Driver, config?: Config) {
+    this.db = db
+    /** Sequence in the properties definitions is IMPORTANT */
+    this.config = {
+      force: false,
+      table: DEFAULT_MIGRATIONS_DIR_NAME,
+      ...config,
+      migrationsPath: path.resolve(
+        config?.migrationsPath ?? DEFAULT_MIGRATIONS_DIR_PATH
+      )
+    }
+  }
+
+  /**
+   * Performs a database migration read operation.
+   *
+   * You can run up and down with `db.exec(migrations[0].up)` or `db.exec(migrations[0].down)`. Or with other commands that can run SQL code.
+   *
+   * @param migrationPath path to migration folder can be relative or absolute. If not found uses current working directory
+   * @returns IMigrate.MigrationData[]. List of objects with name, id of migrations and SQL code for up and down migration.
+   *
+   * Get the list of migrations, for example:
+   *  { id: 1, name: 'initial', filename: '001-initial.sql', up: ..., down: ... }
+   *  { id: 2, name: 'feature', filename: '002-feature.sql', up: ..., down: ... }
+   */
+  readMigrations = async (): Promise<MigrationData[]> => {
```

Comment:
classes in general do not use arrow functions to define methods. there are specific cases where you would use an arrow function, but the methods in this file isn't one of them

Reply:
Okay, I understand what you mean)

It's very interesting to go deep in learning JS and other languages. I understand now that I'm just using some part that I know how it works for some purposes but not for everything because I don't know in the depth the under hood processes